port/linux/ipadapter.c

```diff
@@ -706,9 +717,18 @@ process_interface_change_event(void)
   if (if_state_changed) {
     for (i = 0; i < num_devices; i++) {
       ip_context_t *dev = get_ip_context_for_device(i);
-      oc_network_event_handler_mutex_lock();
-      refresh_endpoints_list(dev);
-      oc_network_event_handler_mutex_unlock();
+      if (dev == NULL) {
+        continue;
+      }
+      bool swapped = false;
+      int expected = OC_ATOMIC_LOAD(dev->flags);
```

Comment:
why you don't call `OC_ATOMIC_LOAD32` because `int` is 32bit ?

Reply:
GCC's `__atomic_` functions work for (correctly aligned) integers of all sizes, so it's not necessary to use integers of specific size. But I can use `int32_t` and `OC_ATOMIC_LOAD32` if you think it would help with clarity.

Reply:
Yes. It is more clear.
Maybe we don't need a similar function as `OC_ATOMIC_LOAD` without suffix 32. Some ports don't support atomic 64bits operations.